src/de/otto/tesla/stateful/configuring.clj

```diff
@@ -24,26 +24,28 @@
 
 (defn- load-properties [name & [type]]
   (cond
-    (and (= :properties type) (io/resource name)) (load-properties-from-resource (io/resource name))
+    (and (= :resource type) (io/resource name)) (load-properties-from-resource (io/resource name))
     (and (= :file type) (.exists (io/file name))) (load-properties-from-resource (io/file name))))
 
 (defn load-config-from-property-files []
-  (let [defaults (load-properties "default.properties" :properties)
+  (let [defaults (load-properties "default.properties" :resource)
         config (load-properties (or (:config-file env/env) "application.properties") :file)
-        local (load-properties "local.properties" :properties)]
+        local (load-properties "local.properties" :resource)]
     (merge defaults config local env/env)))
 
-(defn- load-edn [name]
-  (when-let [resource (io/resource name)]
+(defn- load-edn [name & [type]]
+  (when-let [resource (cond
+                        (= :resource type) (io/resource name)
+                        (and (= :file type) (.exists (io/file name))) (io/file name))]
     (-> resource
         (io/reader)
         (PushbackReader.)
         (read))))
 
 (defn load-config-from-edn-files []
-  (let [defaults (load-edn "default.edn")
-        application (load-edn (or (:config-file env/env) "application.edn"))
-        local (load-edn "local.edn")
+  (let [defaults (load-edn "default.edn" :resource)
+        application (load-edn (or (:config-file env/env) "application.edn") :file)
```

Comment:
Why do you expect the `default.edn` and `local.edn` to be resources but the `application.edn` to be a file?